<launch>
	<arg name="scan_topic" default="/scan"/>
	<arg name="odometry_topic" default="/vesc/odom"/>
	<arg name="max_particles" default="1000"/>

	<!-- Load additional parameters from cfg/params.yaml -->
	<rosparam command="load" file="$(find lab5_localization)/cfg/params.yaml" ns="/particle_filter/" />
	
	<node pkg="lab5_localization" type="particle_filter.py" name="particle_filter_node" output="screen">
		<param name="scan_topic" value="$(arg scan_topic)"/>
		<param name="odometry_topic" value="$(arg odometry_topic)"/>
		<param name="max_particles" value="$(arg max_particles)"/>
		<param name="max_viz_particles" value="200"/>
		<param name="range_method" value="cddt"/>

		<param name="theta_discretization" value="108"/>
		<!-- max sensor range in meters -->
		<param name="max_range" value="10"/> 
	</node>
</launch>
